client/handle_http.go

```diff
@@ -1336,6 +1335,39 @@ func generateTransferDetailsUsingCache(cache CacheInterface, opts transferDetail
 	return nil
 }
 
+// This function gets the amount of caches we will try equal to the configured "cachesToTry". It sets up our transfer attempt details for us and returns it.
+// This function also ensures that we do not try any duplicate caches
+func getCachesToTry(closestNamespaceCaches []CacheInterface, job *TransferJob, cachesToTry int, packOption string) (transfers []transferAttemptDetails) {
+	// The counter for the amount of caches we currently have listed to try
+	cachesListed := 0
+	// This list keeps track of the caches we have seen to ensure we do not have any duplicates
+	cacheList := make(map[CacheInterface]bool)
+	// Caches is just the list of caches we want to try (we print this out in debug logs)
+	caches := make([]CacheInterface, 0)
+
+	for _, cache := range closestNamespaceCaches {
+		// If we listed the amount of caches we want to try, we can end our loop
+		if cachesListed == cachesToTry {
+			break
+		}
+		// if the current cache is already a cache we are trying, skip it
+		if cacheList[cache] {
+			continue
+		} else {
+			cachesListed++
+			caches = append(caches, cache)
+			cacheList[cache] = true
+			td := transferDetailsOptions{
+				NeedsToken: job.namespace.ReadHTTPS || job.namespace.UseTokenOnRead,
+				PackOption: packOption,
+			}
+			transfers = append(transfers, generateTransferDetailsUsingCache(cache, td)...)
+		}
+	}
+	log.Debugln("Trying the caches:", caches)
```

Comment:
@jhiemstrawisc since you are going to review this PR, I know you said you wanted this debug statement gone but IMO I like this debug line. It's nice to see the specific caches we are trying outside of the massive list we get printed from the director. Then later on in the client code, we print out what specific cache we are tackling before we attempt a transfer.

Reply:
As luck would have it, Brian is overhauling most of these error messages in #1301. We'll get your changes in first to make sure his changes are the final say.